src/invariant/ConservationOfLumens.cpp

```diff
@@ -71,7 +79,54 @@ calculateDeltaBalance(LedgerEntry const* current, LedgerEntry const* previous)
     }
 #ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
     case CONTRACT_DATA:
-        break;
+    {
+        auto const& contractData = current ? current->data.contractData()
+                                           : previous->data.contractData();
+        if (contractData.contractID != lumenContractID ||
+            contractData.key.type() != SCV_VEC || !contractData.key.vec() ||
+            contractData.key.vec().size() == 0)
+        {
+            return 0;
+        }
+
+        // The balanceSymbol should be the first entry in the SCVec
+        if (!(contractData.key.vec()->at(0) == balanceSymbol))
+        {
+            return 0;
```

Comment:
Should this be an assert instead?  I.e. if the contract is already a Lumen SAC, is it expected that "Balance" exist and is at the beginning?